src/tests/payments.js

```diff
@@ -0,0 +1,277 @@
+(function() {
+
+var assert = require('assert');
+var eq_ = assert.eq_;
+var contains = assert.contains;
+var mock = assert.mock;
+var defer = require('defer');
+var logger = require('log')('tests/payments');
+
+
+function FakeFxPay() {}
+
+FakeFxPay.prototype.init = function(opt) {
+    opt.oninit();
+};
+
+FakeFxPay.prototype.purchase = function(productId, callback) {
+    callback();
+};
+
+
+function FakeRequests() {
+    this.webpayJWT = this._makeJWT({typ: 'mozilla/payments/pay/v1'});
+    this.promiseToPost = null;
+}
+
+FakeRequests.prototype._makeJWT = function(payRequest) {
+    if (!payRequest) {
+        payRequest = {typ: 'mozilla/payments/pay/v1'};
+    }
+    return '<algo>.' + btoa(JSON.stringify(payRequest)) + '.<sig>';
+};
+
+FakeRequests.prototype.post = function() {
+    // When the Fireplace adapter posts to the API it is generating a
+    // JWT to begin the purchase.
+    var response = {
+        webpayJWT: this.webpayJWT,
+        productId: '<productId>',
+        contribStatusURL: '/transaction/status',
+    };
+    if (!this.promiseToPost) {
+        this.promiseToPost = defer.Deferred().resolve(response).promise();
+    }
+    return this.promiseToPost;
+};
+
+FakeRequests.prototype.get = function() {
+    // When the Fireplace adapter makes a get request it is checking the
+    // status of the transaction.
+    var response = {
+        // The transaction has been verified and has completed successfully.
+        status: 'complete',
+    };
+    return defer.Deferred().resolve(response).promise();
+};
+
+
+function fakeDeps() {
+    return {
+        requests: new FakeRequests(),
+        notification: {notification: function() {}},
+        settings: {},
+    };
+}
+
+
+function PaymentWindow() {
+    this.location = '';
+    this.closed = false;
+    this.close = function() {};
+}
+
+
+function FakeFireplaceWindow() {
+    this.origin = 'https://marketplace.firefox.com';
+    this.open = function() {
+        this.paymentWindow = new PaymentWindow();
+        return this.paymentWindow;
+    };
+    this.addEventListener = function(eventType, handler) {
+        logger.log('calling handler for:', eventType);
+        // Simulate Webpay's postMessage back to fxpay.
+        handler({
+            origin: this.origin,
+            data: {
+                // Indicate that the user payment flow is
+                // finished. Next fxpay will verify the transaction.
+                status: 'ok',
+            }
+        });
+    };
+    this.removeEventListener = function() {};
+}
+
+
+function FakeProduct() {
+    this.slug = 'some-slug';
+    this.payment_required = true;
+}
+
+
+function purchaseOptions(opt) {
+    opt = opt || {};
+    return {
+        fxpaySettings: {
+            mozApps: null,
+            mozPay: null,
+            window: opt.fakeWindow || new FakeFireplaceWindow(),
+            // On Firefox there is some difference in fxpay's scope
```

Comment:
I think it's a bug. I only need to set this for Firefox, not Chrome. I spent time down a rabbit hole on it but couldn't figure it out. I think if we requireJS-ify fxpay it will go away.